samples/net/zperf/src/zperf_udp_uploader.c

```diff
@@ -195,38 +209,44 @@ void zperf_udp_upload(const struct shell *shell,
 	do {
 		struct zperf_udp_datagram *datagram;
 		struct zperf_client_hdr_v1 *hdr;
-		uint32_t loop_time;
+		uint32_t secs, usecs;
+		uint64_t loop_time;
 		int32_t adjust;
 		int ret;
 
 		/* Timestamp */
-		loop_time = k_cycle_get_32();
+		loop_time = z_tick_get();
 
 		/* Algorithm to maintain a given baud rate */
 		if (last_loop_time != loop_time) {
-			adjust = packet_duration - time_delta(last_loop_time,
-							      loop_time);
+			adjust = (int32_t)
+				(packet_duration -
+				 k_ticks_to_us_ceil64(loop_time -
+						      last_loop_time));
 		} else {
 			/* It's the first iteration so no need for adjustment
 			 */
 			adjust = 0;
 		}
 
-		if (adjust >= 0 || -adjust < delay) {
+		if (adjust >= 0 || -adjust < (int32_t)delay) {
```

Comment:
For minimal invocation of undefined behavior (sign/unsigned/conversion rank) I think this needs to be:

```
if (adjust >= 0) {
    delay += adjust
} else if ((uint64_t)-adjust < delay) {
    delay -= (uint64_t)-adjust;
} else {
    delay = 0U; /* ... */
}
```